mautic/mautic
# Notification- added subscribe/unsubscribe support

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Now it's not possible identify what push ids are  subscribe/unsubscribe  on OneSignal. 
This PR added small improvements:
- update enabled status for each push ids based on response in notification CampaignSubscriber onCampaignTriggerAction
- extend notification filter in segments  with new subscribed/unsubscribed option

![image](https://user-images.githubusercontent.com/462477/32394388-40811830-c0dd-11e7-897d-4e52b171a2d2.png)

#### Steps to test this PR:
1.  Create segment with Notification filter set to Yes
2.  Create segment with Notification filter set to Subscribed
3.  Create segment with Notification filter set to Unsubscribe
4. Go to website from different supported notification browsers (Opera, Firefox, Chrome) and akcept notification from browser notification
5. Make same mautic profile  on each browser. For example use Mautic form with email field and send form. Should same contact now everyhwere
6. Create campaign with segment source from step 1
7. In these step you should have in push_ids table 3 rows for your contact id, each enabled. 
8.  Go to page from chrome and turn off notifications for website
(quick view - http://www.ibtimes.co.uk/how-turn-off-facebook-push-notifications-google-chrome-1536670)
7. Send notification from campaign to contact
9. Contact should send  to firefox, opera, but not to chrome
10. In these step you should have in push_ids table 3 rows for your contact id, but just two enabled
11. Go to segments and see if filters works (should have 1 contact in all notification segments)
